network/framework/src/protocols/health_checker/mod.rs

```diff
@@ -109,6 +110,8 @@ pub struct HealthChecker<NetworkClient> {
     ping_failures_tolerated: u64,
     /// Counter incremented in each round of health checks
     round: u64,
+
+    connection_events_injection: Option<tokio::sync::mpsc::Receiver<ConnectionNotification>>,
```

Comment:
nit: why `injection` in the name, and not just `connection_events`?

Reply:
it's only used in test code

Reply:
Added comments here and set_connection_source() noting it's just for unit test usage.